openedx_webhooks/github/dispatcher/actions/utils.py

```diff
@@ -15,3 +23,94 @@ def find_issues_for_pull_request(jira, pull_request_url):
     """
     jql = 'project=OSPR AND cf[10904]="{}"'.format(pull_request_url)
     return jira.search_issues(jql)
+
+
+def _get_latest_commit_for_pull_request(repo_name_full: str, number: int) -> str:
+    """
+    Lookup PR commit details and pull out the SHA of the most recent commit
+    """
+    data = _get_latest_commit_for_pull_request_data(repo_name_full, number)
+    commit = {}
+    if data:
+        commit = data[-1]
+    sha = commit.get('sha')
+    logger.info("CLA: SHA %s", sha)
+    return sha
+
+
+def _get_latest_commit_for_pull_request_data(repo_name_full: str, number: int) -> str:
+    """
+    Lookup the HEAD commit SHA for a pull request
+    """
+    url = f"https://api.github.com/repos/{repo_name_full}/pulls/{number}/commits"
+    logger.info("CLA: GET %s", url)
+    response = get_github_session().get(url)
+    log_check_response(response)
+    data = response.json()
+    logger.info("CLA: GOT %s", data)
+    return data
+
+
+def _get_commit_status_for_cla(url):
+    """
+    Send a GET request to the Github API to lookup the build status
+    """
+    logger.info("CLA: GET %s", url)
+    response = get_github_session().get(url)
+    log_check_response(response)
+    data = response.json()
+    logger.info("CLA: GOT %s %s", url, data)
+    cla_status = [
+        status
+        for status in data
+        if status.get('context') == 'cla'
```

Comment:
Reading between the lines of the GitHub API, the context needs to be unique across all possible checkers.  I see contexts like "jenkins/quality".  We should change "cla" to be more specific to us, like "edx/cla" at least.

Reply:
Done in pending commit.